### PR TITLE
DOC: Use unittest.mock on Python 3.

### PR DESCRIPTION
**Pull request by QuLogic**
As in title.

TODO:
* [N/A] Add unit tests and/or doctests in docstrings
* [N/A] Add docstrings and API docs for any new/modified user-facing classes and functions
* [N/A] New/modified features documented in docs/tutorial.rst
* [?] Changes documented in docs/release.rst
* [x] Docs build locally (e.g., run ``tox -e docs``)
* [x] AppVeyor and Travis CI passes
* [x] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by QuLogic** [comment]
I don't know what coveralls is talking about; didn't change any of those files and the changes should not affect them.

**Comment by jakirkham** [comment]
Have tried to merge with `master` in case that fixes it. Hope that is ok.

**Comment by QuLogic** [comment]
Done.

**Comment by QuLogic** [comment]
I don't know why AppVeyor is stuck.

File: requirements_rtfd.txt [hunk 1]
```diff
@@ -4,5 +4,5 @@ setuptools_scm
 sphinx
 sphinx-issues
 numpydoc
-mock
+mock; python_version < '3.3'
```
Comment (jrbourbeau) [inline requirements_rtfd.txt]:
```suggestion
```

**Comment by QuLogic** [comment]
Updated.

**Comment by jakirkham** [comment]
Thanks for the PR @QuLogic and @jrbourbeau for the review! 😄

**Pull request merged**